### PR TITLE
Make 0e.0 and 0xp0 not crash

### LINKED ISSUE #20385: `std.debug.print("{}\n", .{1e.5});` should not compile
### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

`std.debug.print("{}\n",  .{1e.5});`  => 0e0

### Expected Behavior

Code should not compile.

`1e.5` is not a vaild floating point literal.

Trying this in c gives `error: exponent has no digits`

### PR DESCRIPTION
**Pull request by wooster0**
This fixes those sequences of characters crashing.

Fixes #20385

### CODE REVIEW

**Review by Vexu — APPROVED** [review]

**Review by linusg — CHANGES_REQUESTED** [review]

File: lib/std/zig/number_literal.zig
Comment (wooster0) [reply lib/std/zig/number_literal.zig:114]: This kind of thing can be done in a different PR

Comment (linusg) [reply lib/std/zig/number_literal.zig:114]: It's new code, why would you want to fix that in a _separate_ PR?

Comment (wooster0) [reply lib/std/zig/number_literal.zig:114]: This is basically just style. I don't see the problem. You're not pointing out a bug or anything. Unless a maintainer wants this I don't feel like rerunning the CI for this

Comment (Rexicon226) [reply lib/std/zig/number_literal.zig:114]: That is silly, there is literally no gain from not fixing this. A subsequent PR will *also* run the CI??

Comment (wooster0) [reply lib/std/zig/number_literal.zig:114]: I don't see anything to fix. This is just style.

**Comment by linusg** [comment]
For reviewing maintainers: The comment above has mistakenly been marked as resolved, please take a look at it before merging. Thanks!

**Comment by andrewrk** [comment]
I would have appreciated the suggested cleanup, but thanks for the bug fix regardless.

**Pull request merged**
